flow2ml/Data_Augumentation.py

```diff
@@ -435,3 +435,64 @@ def applyClose(self,classPath):
             cv2.imwrite(classPath+"/ClosedImages/Closed"+image, Closed)
           except Exception as e:
             print(f"Closing operation failed due to {e}")
+
+  def applyThreshold(self,classPath):
+    ''' 
+      Applies thresholding augmentation to all the images in the given folder.
+      Args : 
+        classPath : (string) directory containing images for a particular class.
+    '''
+
+    try:
+      os.mkdir(classPath+"/ThresholdedImages")    
+    except:
+      raise Exception("Unable to create directory for thresholded images.")
+    for image in list(os.listdir(classPath)):
+      
+      # Read image
+      img = cv2.imread(classPath+"/"+image)
+      
+      if self.operations['threshold']['type'] not in ['simple', 'adaptive', 'OTSU']:
+        raise Exception("Invalid threshold operation.")
+      else:
+
+        if self.operations['threshold']['type'] == 'adaptive':
+          if img is not None:
```

Comment:
put this if condition globally, i.e., after line 453

Reply:
> put this if condition globally, i.e., in line 455 with 'and' operator

are you talking about combining 455 and 460 if statements together??

Reply:
if img is not None condition is needed globally, hence combine it with line 455 directly with 'and' operator, hence if nay of conditions doesn't match, thresholding shouldn;t happen